### PR TITLE
Set dedicated activity workers for ParentClose and scanner workflows

### PR DESCRIPTION
**Pull request by pdoerner**
**What changed?**
Setting dedicated activity workers for ParentClosePolicy and Scanner system workflows.
Updated the CallerTypes for activities in those workflows.

**Why?**
For more fine-grained control over priorities and rate limits.
See https://github.com/temporalio/temporal/pull/4854

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/worker/parentclosepolicy/processor.go [hunk 1]
```diff
@@ -91,22 +91,34 @@ func New(params *BootstrapParams) *Processor {
...
-	processorWorker.RegisterActivityWithOptions(ProcessorActivity, activity.RegisterOptions{Name: processorActivityName})
+	if err := processorWorker.Start(); err != nil {
+		return err
+	}
+
+	activityWorker := s.sdkClientFactory.NewWorker(svcClient, processorActivityTaskQueueName, getActivityWorkerOptions(s))
+	activityWorker.RegisterActivityWithOptions(ProcessorActivity, activity.RegisterOptions{Name: processorActivityName})
 
-	return processorWorker.Start()
+	return activityWorker.Start()
```
Comment (yycptt) [inline service/worker/parentclosepolicy/processor.go:104]: if activityWorker failed to start, the workflow worker should be stopped I think.

File: service/worker/parentclosepolicy/workflow.go [hunk 2]
```diff
@@ -50,8 +50,10 @@ import (
 )
 
 const (
-	// processorTaskQueueName is the taskqueue name
+	// processorTaskQueueName is the task queue name for workflow tasks
 	processorTaskQueueName = "temporal-sys-processor-parent-close-policy"
+	// processorActivityTaskQueueName is the task queue name for activity tasks
+	processorActivityTaskQueueName = "temporal-sys-processor-parent-close-policy-activity"
```
Comment (yycptt) [inline service/worker/parentclosepolicy/workflow.go:56]: Is this task queue get used in workflow code?

File: service/worker/parentclosepolicy/processor.go [hunk 3]
```diff
@@ -91,22 +91,34 @@ func New(params *BootstrapParams) *Processor {
...
 		MaxConcurrentWorkflowTaskExecutionSize: p.cfg.MaxConcurrentWorkflowTaskExecutionSize(),
-		MaxConcurrentActivityTaskPollers:       p.cfg.MaxConcurrentActivityTaskPollers(),
 		MaxConcurrentWorkflowTaskPollers:       p.cfg.MaxConcurrentWorkflowTaskPollers(),
-		BackgroundActivityContext:              ctx,
+	}
+}
+
+func getActivityWorkerOptions(p *Processor) worker.Options {
+	ctx := context.WithValue(context.Background(), processorContextKey, p)
+	ctx = headers.SetCallerType(ctx, headers.CallerTypeAPI)
```
Comment (yycptt) [inline service/worker/parentclosepolicy/processor.go:116]: plz use background caller type. parent close policy is not user API requests.

**Review by yycptt — COMMENTED** [review]
It looks like we don't really need this since those two components are already using their own task queue for activities, not the default worker task queue. so we already have separate control for them.

**Comment by pdoerner** [comment]
Discussed offline and decided this is unnecessary, since these components are already using dedicated workers.

**Pull request closed without merging**